### PR TITLE
Add CLI tooling to fetch files from filestore

### PR DESCRIPTION
**Pull request by mattrobenolt**
This is currently super cumbersome to debug any artifacts that have been uploaded.

@getsentry/infrastructure

### CODE REVIEW

File: src/sentry/utils/json.py [hunk 1]
```diff
@@ -63,6 +63,11 @@ def better_default_encoder(o):
 )
 
 
+def dump(value, fp, **kwargs):
```
Comment (tkaemming) [inline src/sentry/utils/json.py]: `**kwargs` aren't used here, seems like it'd be confusing just to drop them silently.

File: src/sentry/runner/commands/files.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    obj = {
+        'id': file.id,
+        'name': file.name,
+        'headers': file.headers,
+        'size': file.size,
+        'sha1': file.checksum,
+        'dateCreated': file.timestamp,
+    }
+
+    stdout = click.get_binary_stream('stdout')
```
Comment (tkaemming) [inline src/sentry/runner/commands/files.py]: Wouldn't this be a text stream?

File: src/sentry/runner/commands/files.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    with file.getfile() as fp:
+        for chunk in fp.chunks():
+            stdout.write(chunk)
+
+
+@files.command()
+@click.argument('id', type=click.INT, metavar='FILE_ID')
+@click.option('--format', default='json', type=click.Choice(('json', 'yaml')))
+@configuration
+def show(id, format):
```
Comment (tkaemming) [inline src/sentry/runner/commands/files.py]: `info` or `details` might be more specific about what's happening here.

File: src/sentry/utils/json.py (hunk 1 above)
Comment (mattrobenolt) [inline src/sentry/utils/json.py]: We do this on all of the other ones to be API compatible, but we are enforcing our own params.

File: src/sentry/runner/commands/files.py (hunk 2 above)
Comment (mattrobenolt) [inline src/sentry/runner/commands/files.py]: Nope. They're bytes coming out. What if the file is a dSYM file or an image? We just want binary.

File: src/sentry/runner/commands/files.py (hunk 3 above)
Comment (mattrobenolt) [inline src/sentry/runner/commands/files.py]: I like `info`. Will change.

File: src/sentry/runner/commands/files.py (hunk 2 above)
Comment (tkaemming) [inline src/sentry/runner/commands/files.py]: Isn't this just writing the metadata, not the contents?

Comment (mattrobenolt) [inline src/sentry/runner/commands/files.py]: Oh, this is under `show`. Yeah, I'll change. Probably doesn't matter much. I thought this was under `get`.

File: src/sentry/utils/json.py (hunk 1 above)
Comment (tkaemming) [inline src/sentry/utils/json.py]: Yeah, I did realize it was probably out of the scope of this review. I still don't think it's a good pattern, though.

**Comment by tkaemming** [comment]
👍

**Pull request merged**